### PR TITLE
[DF] Use enable_if_t alias wherever possible, for readability

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx [hunk 1]
```diff
@@ -509,14 +508,14 @@ public:
    }
 
    // if the fObjects vector type is derived from TObject, return the name of the object
-   template <typename T = HIST, typename std::enable_if<std::is_base_of<TObject, T>::value>::type * = nullptr>
+   template <typename T = HIST, std::enable_if_t<std::is_base_of<TObject, T>::value, int> = 0>
```
Comment (ikabadzhov) [inline tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:511]: shouldn't that be `std::enable_if_t<std::is_base_of<TObject, T>::value>* = nullptr`?

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx [hunk 2]
```diff
@@ -509,14 +508,14 @@ public:
...
    std::string GetActionName()
    {
       return std::string(fObjects[0]->IsA()->GetName()) + "<BR/>" + std::string(fObjects[0]->GetName());
    }
 
    // if fObjects is not derived from TObject, indicate it is some other object
-   template <typename T = HIST, typename std::enable_if<!std::is_base_of<TObject, T>::value>::type * = nullptr>
+   template <typename T = HIST, std::enable_if_t<!std::is_base_of<TObject, T>::value, int> = 0>
```
Comment (ikabadzhov) [inline tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:518]: same as above ^

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:511]: it should make no difference, this form takes less characters and is the one used everywhere else

**Review by ikabadzhov — APPROVED** [review]

**Comment by ikabadzhov** [comment]
btw clang complains

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:511]: > there is no template type or non-type argument there

mmh not quite, the template parameter is of type `int` there. I can explain in person :)

**Comment by eguiraud** [comment]
you mean clang-format I guess (clang is a compiler but there were no compilation errors) -- my version, that I think is newer than the one in the CI, does not see anything. The version in the CI complains about lines that are not directly touched by the PR. Merging.

**Pull request merged**